### PR TITLE
search navigation block

### PR DESCRIPTION
**Pull request by nhart**

### CODE REVIEW

File: includes/blocks.inc [hunk 1]
```diff
@@ -888,3 +896,98 @@ function islandora_solr_sort() {
...
+      $limit = $query_processor->solrLimit;
+      $num_found = $query_processor->islandoraSolrResult['response']['numFound'];
+      if ($num_found > 1 && $current_pid) {
+        foreach ($objects as $key => $object) {
+          if ($object['PID'] == $current_pid) {
+            $current_pid_key = $key;
+            break;
+          }
+        }
+        if ($current_pid_key !== NULL) {
```
Comment (jordandukart) [inline includes/blocks.inc]: There is an issue here. As the Solr results are alterable modules such as the Book SP, https://github.com/Islandora/islandora_solution_pack_book/blob/7.x/islandora_book.module#L199-L236, we can't rely on the PID being grabbed from the URL to be equivalent to the search result.

That being said, you could probably rely upon the offset and store it with your other parameters.

**Comment by jordandukart** [comment]
I think in general we should store these parameters in the $_SESSION so our URLs don't get polluted. You could instead have a singular parameter referring to the id of the set of search results in the URLs. If for some reason these need to be persistent we -could- store them in a database but then we get into privacy concerns.

File: includes/blocks.inc (hunk 1 above)
Comment (nhart) [inline includes/blocks.inc]: good point. for the offset, wouldn't each solr result link need the current position appended to the url along with the search id?

**Comment by nhart** [comment]
appears travis is failing on other work not related to this PR.

I've updated the code to store what I can in the session. Perhaps a different set of eyes can come up with a better solution.

**Comment by nhart** [comment]
Yes. I thought 208 was already dealt with? But I see its still open.

File: includes/admin.inc [hunk 2]
```diff
@@ -373,6 +373,13 @@ function islandora_solr_admin_settings($form, &$form_state) {
     '#description' => t('The number of results to show per page.'),
     '#default_value' => variable_get('islandora_solr_num_of_results', '20'),
   );
+  $form['islandora_solr_tabs']['default_display_settings']['islandora_solr_search_navigation'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Enable search navigation block'),
+    '#size' => 5,
```
Comment (willtp87) [inline includes/admin.inc]: Not used by checkbox https://api.drupal.org/api/drupal/developer!topics!forms_api_reference.html/7#size

File: includes/query_processor.inc [hunk 3]
```diff
@@ -339,6 +339,24 @@ class IslandoraSolrQueryProcessor {
       $datastream_field = variable_get('islandora_solr_datastream_id_field', 'fedora_datastreams_ms');
       $object_label = variable_get('islandora_solr_object_label_field', 'fgs_label_s');
       if (!empty($object_results)) {
+        if (variable_get('islandora_solr_search_navigation', FALSE)) {
+          $id = substr(str_shuffle("0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"), 0, 10);
```
Comment (willtp87) [inline includes/query_processor.inc]: https://api.drupal.org/api/drupal/includes!bootstrap.inc/function/drupal_random_bytes/7 is more normal, can we not use it?

File: theme/islandora-solr-search-navigation-block.tpl.php [hunk 4]
```diff
@@ -0,0 +1,17 @@
+<?php
+/**
+ * @file
+ * Islandora solr search return url for use on object pages
+ *
+ * Variables available:
+ * - $islandora_solr_search_return_link: Link to return solr search.
+ *
+ * @see template_preprocess_islandora_solr_search_return_link()
```
Comment (willtp87) [inline theme/islandora-solr-search-navigation-block.tpl.php]: Where is this function to see?

File: includes/query_processor.inc (hunk 3 above)
Comment (nhart) [inline includes/query_processor.inc]: yeh.. i just pasted that quickly to get it working. I will update it

**Comment by willtp87** [comment]
https://github.com/Islandora/islandora_solr_search/pull/211/files#diff-ed5c200e98e6d37a085ea47610c9b9f1R366 should probably be removed or edited, as we now have two initializations of the var.

**Comment by nhart** [comment]
not sure I follow you here. https://github.com/Islandora/islandora_solr_search/pull/211/files#diff-ed5c200e98e6d37a085ea47610c9b9f1R366

you want me to assign the the drupal var to a php var to be later used in the loop?

File: includes/blocks.inc [hunk 5]
```diff
@@ -888,3 +896,124 @@ function islandora_solr_sort() {
...
+ * Search results navigation block.
+ *
+ * Pull params out of the URL to determine the link to return
+ * the original search, and the previous and next object in the search
+ * according to the current object.
+ *
+ * @return string
+ *   The rendered html for the block content.
+ */
+function islandora_solr_search_navigation() {
```
Comment (willtp87) [inline includes/blocks.inc:912]: Talked with Nelson, he's going to add some inline docs to clarify how this function works.

**Comment by willtp87** [comment]
$object_results[$object_index]['object_url_params'] is being set twice, it doesn't have to be. (in regards to https://github.com/Islandora/islandora_solr_search/pull/211#issuecomment-70149362)

**Pull request merged**